### PR TITLE
Adding README, CONTRIBUTING, LeanSatTactic

### PR DESCRIPTION
**Pull request by nwetzler**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- Illustrative, minimal, and reproducible test cases
+- Commit hash of leanbv that you used
+- Suggested modifications, if any
+
+### Licensing
+
+See the
+[LICENSE](https://github.com/leanprover/leanbv/blob/main/LICENSE) file
+for leanbv's licensing. All files in this project will be copyright
+only by "Amazon.com, Inc. or its affiliates". The PR template will
```
Comment (shigoel) [inline CONTRIBUTING.md:32]: The PR template doesn't exist here yet. Here's an example template from LNSym:
https://github.com/leanprover/LNSym/blob/main/.github/pull_request_template.md

File: README.md [hunk 2]
```diff
@@ -1 +1,82 @@
...
+[lean-auto](https://github.com/leanprover-community/lean-auto) allow
+Lean to appeal to SMT, but this trusts that the SMT solver is
+correct. Validating SMT proofs is a research area, but SAT proofs are
+well studied. Moreover, many program verification problems, like
+correctness of block ciphers and hashes, need no theories beyond
+bit-vectors, and are well suited to automation through SAT.
+
+`leanbv` is a start at a tactic for Lean that can convert proof goals
+into SAT problems, invoke a solver, and then soundly interpret SAT and
+UNSAT results back into the Lean without the a trusted assumption.
```
Comment (shigoel) [inline README.md:25]: typo: "the a"

Comment (nwetzler) [reply README.md]: Done.

File: CONTRIBUTING.md (hunk 1 above)
Comment (nwetzler) [reply CONTRIBUTING.md:32]: Added the pull request template.

**Review by shigoel — APPROVED** [review]
LGTM!

**Pull request merged**